src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -131,6 +140,149 @@ private static void addDistanceGraphEdge(
+      log.error("Error adding 2 winding transformer edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    try {
+      rawGridElements.getTransformer3Ws().forEach(trafo3w -> addImpedanceGraphEdge(graph, trafo3w));
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding 3 winding transformer edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    // if we reached this point, we can safely return a valid graph
+    return Optional.of(graph);
+  }
+
+  /**
+   * Adds an {@link ImpedanceWeightedEdge} to the provided graph between the provided nodes a and b.
+   * By implementation of jGraphT this side effect cannot be removed. :(
+   *
+   * @param graph the graph to be altered
+   * @param connectorInput the connector input element
+   */
+  private static void addImpedanceGraphEdge(
+      ImpedanceWeightedGraph graph, ConnectorInput connectorInput) {
+    NodeInput nodeA = connectorInput.getNodeA();
+    NodeInput nodeB = connectorInput.getNodeB();
+    /* Add an edge if it is not a switch or the switch is closed */
+    if (!(connectorInput instanceof SwitchInput) || ((SwitchInput) connectorInput).isClosed())
+      graph.addEdge(nodeA, nodeB);
+
+    if (connectorInput instanceof LineInput) {
+      LineInput line = (LineInput) connectorInput;
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          calcImpedance(line.getType().getR(), line.getType().getX(), line.getLength()));
+    }
+    if (connectorInput instanceof SwitchInput) {
+      SwitchInput sw = (SwitchInput) connectorInput;
+      // assumption: closed switch has a resistance of 1 OHM
+      if (sw.isClosed()) graph.setEdgeWeight(graph.getEdge(nodeA, nodeB), 1);
+    }
+    if (connectorInput instanceof Transformer2WInput) {
+      Transformer2WInput trafo2w = (Transformer2WInput) connectorInput;
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          sqrt(
+              pow(trafo2w.getType().getrSc().to(OHM).getValue().doubleValue(), 2)
+                  + pow(trafo2w.getType().getxSc().to(OHM).getValue().doubleValue(), 2)));
+    }
+    if (connectorInput instanceof Transformer3WInput) {
+      Transformer3WInput trafo3w = (Transformer3WInput) connectorInput;
+      graph.addEdge(nodeA, trafo3w.getNodeC());
+
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, nodeB),
+          sqrt(
+              pow(
+                      trafo3w.getType().getrScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getrScB().to(OHM).getValue().doubleValue(),
+                      2)
+                  + pow(
+                      trafo3w.getType().getxScA().to(OHM).getValue().doubleValue()
+                          + trafo3w.getType().getxScB().to(OHM).getValue().doubleValue(),
+                      2)));
+
+      graph.setEdgeWeight(
+          graph.getEdge(nodeA, trafo3w.getNodeC()),
```

Comment:
Not sure how this internally works but do we need an edge from `nodeB` to `nodeC` as well? 
Because otherwise it seems to me that to get from  c to b we have to go from c to a and then from a to c which raises the impedance artificially.